src/avk.cpp

```diff
@@ -8235,6 +8367,175 @@ namespace avk
+				}
+			}
+#endif //_DEBUG
+			const bool detectExtent = !aRenderAreaExtent.has_value();
+			std::vector<vk::RenderingAttachmentInfoKHR> colorAttachments = {};
+			std::optional<vk::RenderingAttachmentInfoKHR> depthAttachment = {};
+			std::optional<vk::RenderingAttachmentInfoKHR> stencilAttachment = {};
+			// First parse all the attachments into vulkan structs
+			for(uint32_t attachmentIndex = 0; attachmentIndex < aAttachments.size(); attachmentIndex++)
+			{
+				const auto & currAttachment = aAttachments.at(attachmentIndex);
+				const auto & currImageView = aImageViews.at(attachmentIndex);
+				if(detectExtent && !aRenderAreaExtent.has_value())
+				{
+					const auto imageExtent = currImageView->get_image().create_info().extent;
+					aRenderAreaExtent = vk::Extent2D{
+						imageExtent.width - static_cast<uint32_t>(aRenderAreaOffset.x),
+						imageExtent.height - static_cast<uint32_t>(aRenderAreaOffset.y)
+					};
+				}
+#ifdef _DEBUG
+				else if(detectExtent)
+				{
+					const auto imageExtent = currImageView->get_image().create_info().extent;
+					const auto currAreaExtent = vk::Extent2D{
+						imageExtent.width - static_cast<uint32_t>(aRenderAreaOffset.x),
+						imageExtent.height - static_cast<uint32_t>(aRenderAreaOffset.y)
+					};
+					if(currAreaExtent != aRenderAreaExtent.value())
+					{
+						throw avk::runtime_error("Autodetect extent failed because the images passed in image views have differing extents");
+					}
+				}
+#endif //_DEBUG
+				if(currAttachment.is_used_as_color_attachment())
+				{
+					colorAttachments.push_back(
+						vk::RenderingAttachmentInfoKHR{}
+							.setImageView(currImageView->handle())
+							.setImageLayout(vk::ImageLayout::eColorAttachmentOptimal)
+							// TODO(msakmary) add support for resolve and MSAA
+							// .setResolveMode()
+							// .setResolveImageView()
+							// .setResolveImageLayout()
+							.setLoadOp(to_vk_load_op(currAttachment.mLoadOperation.mLoadBehavior))
+							.setStoreOp(to_vk_store_op(currAttachment.mStoreOperation.mStoreBehavior))
+							.setClearValue(vk::ClearColorValue(currAttachment.clear_color()))
+					);
+				} 
+				else // currAttachment is either used as depth or as stencil
+				{
+					// TODO(msakmary): This will brake if we want depth image and stencil both D24S8 but separate images (so two D24S8 images 
+					//                 one used as depth one as stencil) probably should have this info in a custom attachment type.
+					//                 I think something like begin_rendering_attachment should be added which would have an explicit field 
+					//                 which would denote how to use the attachment - use this attachment as stencil, depth or color
+					if(is_depth_format(currAttachment.format()))
+					{
+						if(depthAttachment.has_value())

```

Comment:
Yeah, more options are always a good thing. Please proceed as you have planned!

Reply:
This would get resolved by the change I propose in [this comment](https://github.com/cg-tuwien/Auto-Vk/pull/92#issuecomment-2016222783). I could use the `dynamic_usage` of each attachment to alleviate this issue.